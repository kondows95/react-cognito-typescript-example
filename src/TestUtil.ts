import {ReactElement} from 'react';
import { render, fireEvent, RenderResult } from '@testing-library/react';
export default class TestUtil
{
  public render: RenderResult;
  constructor(component:ReactElement) {
    this.render = render(component);
  }

  setValue(testId: string, value: string | number) {
    const elem = this.render.getByTestId(testId);
    fireEvent.change(elem, {target: {value: value}})
  }

  click(testId: string) {
    const elem = this.render.getByTestId(testId);
    fireEvent.click(elem);
  }

  getAll(testId: string) {
    return this.render.getAllByTestId(testId);
  }

  get(testId: string) {
    return this.render.getByTestId(testId);
  }
}