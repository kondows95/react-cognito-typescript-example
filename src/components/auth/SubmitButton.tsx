import React, {PropsWithChildren} from 'react';
import { Button, CircularProgress}  from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
  progress: {
    position: 'absolute',
    top: '50%',
    left: '50%',
    marginTop: -12,
    marginLeft: -12,
  },
  item: {
    [theme.breakpoints.down("xs")]: { 
      padding: theme.spacing(1, 0, 1), //TXL
    }, 
    [theme.breakpoints.up("sm")]: { 
      padding: theme.spacing(2, 0, 1, 1), //TRBL
    },
  },
}));

type Props = { disabled: boolean }

const SubmitButton: React.FC<PropsWithChildren<Props>> = (props) => {
  const classes = useStyles();

  let loading = null;
  if (props.disabled) {
    loading = <CircularProgress size={20} color="primary" className={classes.progress} />;
  }
  return (
    <Button
        fullWidth 
        {...props} 
        variant="contained" 
        color="primary"
        type="submit"
      > 
        {props.children || "Submit"}
        {loading}
      </Button>
  )
};
export default SubmitButton;