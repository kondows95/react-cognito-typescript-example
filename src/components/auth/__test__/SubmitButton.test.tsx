import React from 'react';
import SubmitButton from '../SubmitButton';
import '@testing-library/jest-dom/extend-expect';
import TestUtil from '../../../TestUtil'; 
let util: TestUtil;

describe("Enable button",()=>{
  beforeEach(() => {
    util= new TestUtil(
      <SubmitButton　disabled={false} data-testid="submitButton">
        enable button
      </SubmitButton>
    );
  });

  it('Snapshot', () => {
    expect(util.render.asFragment()).toMatchSnapshot();
  });

  it ('button is enabled', () => {
    const elem = util.get("submitButton") as HTMLButtonElement;
    expect(elem.disabled).toBe(false);
  });

  it ('children displayed', () => {
    expect(util.render.container).toHaveTextContent('enable button'); 
  });
});

describe("Dispable button",()=>{
  beforeEach(() => {
    util= new TestUtil(
      <SubmitButton　disabled={true} data-testid="submitButton">
        disable button
      </SubmitButton>
    );
  });

  it('Snapshot', () => {
    expect(util.render.asFragment()).toMatchSnapshot();
  });

  it ('button is disabled', () => {
    const elem = util.get("submitButton") as HTMLButtonElement;
    expect(elem.disabled).toBe(true);
  });

  it ('children displayed', () => {
    expect(util.render.container).toHaveTextContent('disable button'); 
  });
});

describe("No children",()=>{
  beforeEach(() => {
    util= new TestUtil(
      <SubmitButton　disabled={true} data-testid="submitButton" />
    );
  });

  it('Snapshot', () => {
    expect(util.render.asFragment()).toMatchSnapshot();
  });

  it ('label is Submit', () => {
    expect(util.render.container).toHaveTextContent('Submit'); 
  })
});

