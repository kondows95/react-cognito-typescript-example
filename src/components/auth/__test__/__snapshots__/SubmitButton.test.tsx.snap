// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dispable button Snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled MuiButton-fullWidth Mui-disabled"
    data-testid="submitButton"
    disabled=""
    tabindex="-1"
    type="submit"
  >
    <span
      class="MuiButton-label"
    >
      disable button
      <div
        class="MuiCircularProgress-root makeStyles-progress-130 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 20px; height: 20px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </span>
  </button>
</DocumentFragment>
`;

exports[`Enable button Snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
    data-testid="submitButton"
    tabindex="0"
    type="submit"
  >
    <span
      class="MuiButton-label"
    >
      enable button
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`No children Snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled MuiButton-fullWidth Mui-disabled"
    data-testid="submitButton"
    disabled=""
    tabindex="-1"
    type="submit"
  >
    <span
      class="MuiButton-label"
    >
      Submit
      <div
        class="MuiCircularProgress-root makeStyles-progress-265 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 20px; height: 20px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </span>
  </button>
</DocumentFragment>
`;
